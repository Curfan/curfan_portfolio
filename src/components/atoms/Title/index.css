.title {
	font-size: 64px;
	line-height: 78px;
	letter-spacing: 0.05em;

	&.subtitle {
		font-size: 36px;
		line-height: 44px;
		letter-spacing: 0.05em;
	}

	& > .tail {
		margin-left: 10px;
	}
}
