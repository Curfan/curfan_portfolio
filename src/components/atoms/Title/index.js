import React from 'react';
import classnames from 'classnames';

import TailIcon from 'images/icon/tail.inline.svg';

import styles from './index.css';

const Title = ({ className, subtitle, withTail, children }) => (
	<div className={classnames(styles.title, className, { [styles.subtitle]: subtitle })}>
		{children}
		{withTail && <TailIcon className={styles.tail} />}
	</div>
);

export default Title;
