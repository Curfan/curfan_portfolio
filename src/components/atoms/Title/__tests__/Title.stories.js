import React from 'react';
import { storiesOf } from '@storybook/react';

import Title from 'components/atoms/Title';

const stories = storiesOf('atoms/Title', module);

stories.add('__interactive', () => <Title>Title</Title>);

stories.add('subtitle', () => <Title subtitle>Title</Title>);

stories.add('with tail', () => (
	<Title subtitle withTail>
		Title
	</Title>
));
