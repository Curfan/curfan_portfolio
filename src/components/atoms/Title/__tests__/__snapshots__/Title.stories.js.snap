// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    Title
  </div>
</DocumentFragment>
`;

exports[`subtitle 1`] = `
<DocumentFragment>
  <div
    class="title subtitle"
  >
    Title
  </div>
</DocumentFragment>
`;

exports[`with tail 1`] = `
<DocumentFragment>
  <div
    class="title subtitle"
  >
    Title
    <test-file-stub
      classname="tail"
    />
  </div>
</DocumentFragment>
`;
